### PR TITLE
👷🏻‍♂️ Add margin checking util

### PR DESCRIPTION
**Pull request by JaredBorders**
# Audit Finding(s)

## FreeMargin check can be made into internal function
In Account.sol, there are multiple checks to ensure that if there is sufficient freeMargin() to allow a trade to occur. These checks are found in functions, _modifyAccountMargin, _perpsV2ModifyMargin, _placeConditionalOrder, and _imposeFee, each are very similar or identical to each other and can be replaced with an internal function.

&&

##  "Duplicate" calls to transferMargin()
In Account.sol’s `_perpsV2ModifyMargin()` there is a branch if `_amount` is greater than zero:

```solidity
if (_amount > 0) {
    if (uint256(_amount) > freeMargin()) {
        revert InsufficientFreeMargin(freeMargin(), uint256(_amount));
    } else {
        IPerpsV2MarketConsolidated(_market).transferMargin(_amount);
    }
} else {
    IPerpsV2MarketConsolidated(_market).transferMargin(_amount);
}
```

Each branch calls the same function `transferMargin()` on the market with the same `_amount` parameter.

### Further details
Added both of these findings/fixes in one PR as they are directly related

### CODE REVIEW

File: src/Account.sol [hunk 1]
```diff
@@ -436,19 +436,15 @@ contract Account is IAccount, Auth, OpsReady {
             });
         } else if (_amount < 0) {
             // if amount is negative, withdraw
-            if (_abs(_amount) > freeMargin()) {
-                /// @dev make sure committed margin isn't withdrawn
-                revert InsufficientFreeMargin(freeMargin(), _abs(_amount));
-            } else {
-                /// @dev failed Synthetix asset transfer will revert and not return false if unsuccessful
-                MARGIN_ASSET.transfer(owner, _abs(_amount));
+            _sufficientMargin({_marginOut: _amount});
```
Comment (JChiaramonte7) [inline src/Account.sol:439]: Seems a bit excessive to use named params for single arguments.

**Review by JChiaramonte7 — APPROVED** [review]
LGTM

File: src/Account.sol (hunk 1 above)
Comment (JaredBorders) [reply src/Account.sol:439]: agreed. ngl this was copilot copying my style lol

Comment (JaredBorders) [reply src/Account.sol]: removed

**Pull request merged**